packages/workers-utils/src/construct-wrangler-config.ts

```diff
@@ -16,103 +19,86 @@ type RoutesRes = {
 	script: string;
 }[];
 
-type CustomDomainsRes = {
-	id: string;
-	zone_id: string;
-	zone_name: string;
-	hostname: string;
-	service: string;
-	environment: string;
-	cert_id: string;
-}[];
+interface APIWorkerConfig {
+	/* sourced from https://developers.cloudflare.com/api/resources/workers/subresources/scripts/methods/list/ */
+	name: string; // property renamed from `id`...
+	entrypoint: string;
+	tags: string[];
+	compatibility_date: string;
+	compatibility_flags: string[];
+	logpush: boolean | undefined;
+	routes: RoutesRes;
+	tail_consumers: TailConsumer[] | undefined;
+	migration_tag?: string;
 
-type WorkerSubdomainRes = {
-	enabled: boolean;
-	previews_enabled: boolean;
-};
-type CronTriggersRes = {
-	schedules: {
-		cron: string;
-		created_on: Date;
-		modified_on: Date;
-	}[];
-};
+	/* sourced from https://developers.cloudflare.com/api/resources/workers/subresources/domains/methods/list/ */
+	domains: Cloudflare.Workers.Domain[] /* sourced from https://developers.cloudflare.com/api/resources/workers/subresources/scripts/subresources/schedules/methods/get/ */;
+	schedules: Cloudflare.Workers.Scripts.Schedules.ScheduleGetResponse.Schedule[];
 
-export interface FullWorkerConfig {
+	/* sourced from https://developers.cloudflare.com/api/resources/workers/subresources/beta/subresources/workers/subresources/versions/methods/get/ using `{version_id}` of `latest` */
+	assets?: AssetConfig;
 	bindings: WorkerMetadata["bindings"];
-	routes: RoutesRes;
-	customDomains: CustomDomainsRes;
-	subdomainStatus: WorkerSubdomainRes;
-	serviceEnvMetadata: ServiceMetadataRes["default_environment"];
-	cronTriggers: CronTriggersRes;
+	observability: Cloudflare.Workers.Beta.Worker.Observability | undefined;
+	limits: { cpu_ms: number } | undefined;
+	placement: Cloudflare.Workers.Beta.Workers.Version.Placement | undefined;
+	subdomain: {
+		enabled: boolean;
+		previews_enabled: boolean;
+	};
 }
 
-function convertWorkerToWranglerConfig(
-	workerName: string,
-	entrypoint: string,
-	{
-		bindings,
-		routes,
-		customDomains,
-		subdomainStatus,
-		serviceEnvMetadata,
-		cronTriggers,
-	}: FullWorkerConfig
-): RawConfig {
-	const mappedBindings = mapWorkerMetadataBindings(bindings);
+function convertWorkerToWranglerConfig(config: APIWorkerConfig): RawConfig {
+	const mappedBindings = mapWorkerMetadataBindings(config.bindings);
 
-	const durableObjectClassNames = bindings
+	const durableObjectClassNames = config.bindings
 		.filter((binding) => binding.type === "durable_object_namespace")
```

Comment:
I think we want to filter out any DO bindings implemented by a different `script_name`, or else we'll include a migration for a class not exported by this script.